pyiceberg/typedef.py

```diff
@@ -199,3 +199,7 @@ def __repr__(self) -> str:
     def record_fields(self) -> List[str]:
         """Return values of all the fields of the Record class except those specified in skip_fields."""
         return [self.__getattribute__(v) if hasattr(self, v) else None for v in self._position_to_field_name]
+
+    def __hash__(self) -> int:
+        """Return hash value of the Record class."""
+        return hash(str(self))
```

Comment:
I think since `__repr_` is defined, the str() might still work? I tested:
```
    r1 = Record(1,2)
    r2 = Record(x=1, y="string value")
    print("----")
    print(str(r1), hash(str(r1)))
    print(str(r2), hash(str(r2)))
```

prints:
```
Record[field1=1, field2=2] -7504199255027864703
Record[x=1, y='string value'] -4897332691101137012
```

Reply:
Hmm, that looks good 👍  I'm always seeing empty records in PyCharm